src/dex/aave-v2/aave-v2.ts

```diff
@@ -0,0 +1,296 @@
+
+import { IDex } from '../../dex/idex';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+
+import { SimpleExchange } from '../simple-exchange';
+import { AaveV2Config, Adapters } from './config';
+import { isAaveV2Pair } from './tokens';
+
+const aaveLendingPool: { [network: string]: string } = {
+  [Network.MAINNET]: '0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9',
+  [Network.POLYGON]: '0x8dff5e27ea6b7ac08ebfdf9eb090f32ee9a30fcf',
+  [Network.AVALANCHE]: '0x4F01AeD16D97E3aB5ab2B501154DC9bb0F1A5A2C',
+};
+
+const WETH_GATEWAY: any = {
+  [Network.MAINNET]: '0xDcD33426BA191383f1c9B431A342498fdac73488',
+  [Network.POLYGON]: '0xbEadf48d62aCC944a06EEaE0A9054A90E5A7dc97',
+  [Network.AVALANCHE]: '0x8a47F74d1eE0e2edEB4F3A7e64EF3bD8e11D27C8',
+};
+
+const WETH_GATEWAY_ABI: any = {
+  [Network.MAINNET]: WETH_GATEWAY_ABI_MAINNET,
+  [Network.POLYGON]: WETH_GATEWAY_ABI_POLYGON,
+  [Network.AVALANCHE]: WETH_GATEWAY_ABI_AVALANCHE,
+};
+
+const REF_CODE = 1;
+
+const Aave2ETHGasCost = 246 * 100;
+const Aave2LendingGasCost = 328 * 1000;
+
+export class AaveV2
+  extends SimpleExchange
+  implements IDex<AaveV2Data, AaveV2Param>
+{
+  readonly hasConstantPriceLargeAmounts = false;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(AaveV2Config);
+
+  logger: Logger;
+
+  private aavePool: Interface;
+  private wethGateway: Interface;
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper, // TODO: add any additional optional params to support other fork DEXes
+  ) {
+    super(dexHelper.augustusAddress, dexHelper.provider);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.wethGateway = new Interface(WETH_GATEWAY_ABI[network]);
+    this.aavePool = new Interface(AAVE_LENDING_POOL_ABI_V2);
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {}
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return Adapters[this.network][side];
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    const tokenAddress = [
+      srcToken.address.toLowerCase(),
+      destToken.address.toLowerCase(),
+    ]
+      .sort((a, b) => (a > b ? 1 : -1))
+      .join('_');
+    return [`${this.dexKey}_${tokenAddress}`];
```

Comment:
Would be better to check the pair is valid Aave pair (if not return `[]`) and use the `${this.dexKey}_${aTokenAddress}` as the pool identifier.  Same as for Aave V1 except the tokens need wrapped first.